### PR TITLE
Move Common Auto-Place Feature to diagram-js

### PR DESCRIPTION
**Pull request by philippfromme**
Requires https://github.com/bpmn-io/diagram-js/pull/453
Requires https://github.com/bpmn-io/diagram-js/pull/459

Related to https://github.com/bpmn-io/dmn-js/pull/492

### CODE REVIEW

**Comment by barmac** [comment]
There is a merge conflict @philippfromme

**Comment by philippfromme** [comment]
@barmac Let me know if there's anything that needs to be changed.

File: lib/features/auto-place/BpmnAutoPlaceUtil.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  if (is(element, 'bpmn:TextAnnotation')) {
+    return getTextAnnotationPosition(source, element);
+  }
+
+  if (isAny(element, [ 'bpmn:DataObjectReference', 'bpmn:DataStoreReference' ])) {
+    return getDataElementPosition(source, element);
+  }
+
+  if (is(element, 'bpmn:FlowNode')) {
+    return getFlowNodePosition(source, element);
```
Comment (barmac) [inline lib/features/auto-place/BpmnAutoPlaceUtil.js:37]: How about if we make a fallback option out of this? Thus, I expect we could avoid problems with custom elements.

Comment (barmac) [reply lib/features/auto-place/BpmnAutoPlaceUtil.js:37]: unless there is a fallback in diagram-js, but I could not find it.

Comment (barmac) [reply lib/features/auto-place/BpmnAutoPlaceUtil.js:37]: OK I got it, sorry for bothering!

**Pull request merged**